utils/gyb_syntax_support/ExprNodes.py

```diff
@@ -23,10 +23,12 @@
          element='TupleElement'),
 
     Node('ArrayElementList', kind='SyntaxCollection',
-         element='ArrayElement'),
+         element='Syntax', element_name='Element',
+         element_choices=['ArrayElement', 'IfConfigDecl', 'PoundSourceLocation']),
```

Comment:
Do we really need `PoundSourceLocation` here?

Reply:
That was a suggestion that came up in the Jira https://bugs.swift.org/browse/SR-8743.

Reply:
hmm, ok. If we want to allow #sourcelocation appears in anywhere in the source file, we probably should define a trivia kind for it rather than putting them in the syntax tree structure.

Reply:
I’ve amended the commit and raised a new Jira https://bugs.swift.org/browse/SR-8800 for which I’m trying to work out the description. I’m not trying to get #sourceLocation working in every context but just for collections which was quite easy to implement an I can see would be useful for code generators. You’re right though, perhaps Syntax should treat it as Trivia.

Reply:
Thank you! @johnno1962